src/osqp_api.c

```diff
@@ -1273,28 +1288,42 @@ void csc_set_data(OSQPCscMatrix* M,
 * Derivative functions
 ****************************/
 #ifdef OSQP_ENABLE_DERIVATIVES
-OSQPInt osqp_adjoint_derivative(OSQPSolver*  solver,
-                                OSQPFloat*     dx,
-                                OSQPFloat*     dy_l,
-                                OSQPFloat*     dy_u,
-                                OSQPCscMatrix* dP,
-                                OSQPFloat*     dq,
-                                OSQPCscMatrix* dA,
-                                OSQPFloat*     dl,
-                                OSQPFloat*     du) {
-
-    OSQPInt status = adjoint_derivative(
+OSQPInt osqp_adjoint_derivative_compute(OSQPSolver*    solver,
+                                        OSQPFloat*     dx,
+                                        OSQPFloat*     dy_l,
+                                        OSQPFloat*     dy_u) {
+
```

Comment:
We should add this code to the beginning of the three derivative functions to make sure they don't crash if called before setup.

```
  // Check if solver has been initialized
  if (!solver || !solver->work || !solver->work->derivative_data)
    return osqp_error(OSQP_WORKSPACE_NOT_INIT_ERROR);
```

Reply:
Done